### DISCUSSION TITLE
How does the licensing work for AFFiNE?

### DISCUSSION THREAD

**Discussion by ssddanbrown** [toeverything/AFFiNE#5947] [General]
Hello :wave:,
I [queried this on Reddit](https://www.reddit.com/r/selfhosted/comments/1b0dyqt/comment/ks7o2z5/) but it never reached a clear answer, and that conversation meandered a bit, so I thought it be best to query this officially here with more detail/context.

I'm trying to understand if it's possible to use AFFiNE using open source code alone, without any required code using a non-[OSD](https://opensource.org/osd) adhering license, which I'd expect would be possible since this is being portrayed as an open source application via the website/repo/marketing.

While most of this repo is under the MIT license, [according to your license](https://github.com/toeverything/AFFiNE/blob/3432f355b0e31637a766f2c81f0d0dcbb1967345/LICENSE#L5) the `packages/backend/server` directory is provided under [a different license](https://github.com/toeverything/AFFiNE/blob/3432f355b0e31637a766f2c81f0d0dcbb1967345/packages/backend/server/LICENSE).
This terms defined in this "EE" license wouldn't generally be considered open source, due to limitations put upon use, modification and distribution.

The license does have the following caveat:

> This EE License applies only to the part of this Software that is not distributed as part of AFFiNE Community Edition (CE). Any part of this Software distributed as part of AFFiNE CE or is served client-side as an image, font, cascading stylesheet (CSS), file which produces or is compiled, arranged, augmented, or combined into client-side JavaScript, in whole or in part, is copyrighted under the MPL2.0 license.

So it's possible that some of this code (or even all of it) may be provided under the MPL2.0 license (which I assume to be the [Mozilla Public License v2.0](https://www.mozilla.org/en-US/MPL/2.0/)), but this is dependent on what is distributed in "AFFiNE Community Edition (CE)". 

Based upon the above, my questions are as follows:

- Where can I find what's considered to officially be part of "AFFiNE Community Edition (CE)"?
- Can AFFiNE be ran from its ([OSD](https://opensource.org/osd) adhering) open source code alone?
- Or have I misunderstood how this `server` package is used? and it's not actually required at all to run/use the core functions of AFFiNE?
  - Based on the Reddit conversation, is sounds like this is required but I've put this question here also in the even I've misunderstood.

I'm specifically stating "[OSD](https://opensource.org/osd) adhering" since that's what many consider to be the definition of open source, but if you've defined your license and marketing based on some other definition that would be good to understand, and help add context to answers to my above queries.

**Comment by EYHN** [comment]
Hey @ssddanbrown, sorry for the late reply. I just had a discussion with our team members and came up with the following response to your question.

The LICENSE in the server part needs to be modified from MPL 2.0 to MIT. This should be our mistake.

* Where can I find what's considered to officially be part of "AFFiNE Community Edition (CE)"?

Folders other than `packages/backend/server` will be AFFiNE Community Edition (CE).
The folder `packages/backend/server` is not a part of AFFiNE, but a part of AFFiNE cloud.
And it's placed under the AFFiNE monorepo to speed up development.


* Can AFFiNE be ran from its (OSD adhering) open source code alone?

Yes, you can build and run AFFiNE from the open source code under OSD definition alone.
Binaries under the release page are built from the open source code under the OSD definition.


* Or have I misunderstood how this `server` package is used? and it's not actually required at all to run/use the core functions of AFFiNE?

* Based on the Reddit conversation, is sounds like this is required but I've put this question here also in the even I've misunderstood.

It's not required and it's an addon. You can still use AFFiNE offline without any issue.
But in this case, sync, collaboration features will not be available. You can connect your client to our AFFiNE cloud to enable these features.
Or alternatively, running proprietary server by yourself to enable these features under the EE terms.

**Comment by ssddanbrown** [comment]
@EYHN Okay, that makes sense, thanks for the response!

It might be best to completely remove that part of the license in regard to "AFFiNE Community Edition (CE)" if you no longer provide a clear specific offering under that exact name, since otherwise it'll be somewhat open to interpretation, and since this caveat doesn't really matter if nothing within that folder is part of the "AFFiNE Community Edition (CE)".

As you develop this, I think it's going to be important in how you make clear what's actually provided as open source or not. Just looking at the website or repo readme, it's hard to understand what parts are actually open source. As an example, on your website homepage, you have the text: `Open-Source Code for Trust and Collaboration` and then, directly below that: `Free for individuals, commercial and team usage fees apply.` which are somewhat somewhat contradictory.

In your Reddit post, you advertised it as a "notion & miro open source alternative" in the title, but then the thing you're actually linking to (docker setup), from what I can tell (by looking in the provided container) is directly running, and relying upon, your non-open-source (enterprise) code.

I respect your right to choose licensing which works best for you, but I also think transparency and clarity is important with these things to avoid users being burnt by thinking they're relying on something open source which actually isn't.

**Comment by forresthopkinsa** [comment]
Has this situation changed at all in the past year? Am I correctly understanding that _all_ server sync, collaboration, and backend functions are not OSI open-source?

**Comment by armync** [comment]
@EYHN Any update on this? 

> The LICENSE in the server part needs to be modified from MPL 2.0 to MIT. This should be our mistake.

**Comment by sontheweb** [comment]
Hey, any updates on this? Can we write our own bolt-on to host it on our cloud and add more collaboration features integrated into our app? I am still very confused about what is OpenSource and what will require us a license. 

Can you please clarify for what the license is required and whom can I contact for it? Do we still need a license if we will be completely rewriting the collaboration features and self-hosting on cloud?

**Comment by atomlab** [comment]
Hi @EYHN and AFFiNE team,

Thanks for the detailed response in this thread and for keeping AFFiNE so community-focused! I’m just curious about a few specifics regarding the self-hosted (Local FOSS) version and its integration with Cloud Basic, as mentioned on the pricing page.
From your reply, I understand that the core AFFiNE codebase is fully open under MIT (pending the license fix for `packages/backend/server`), and the server part relates to AFFiNE Cloud. However, for the self-hosted version with Cloud Basic, there are limits like 10 members per workspace, 100 GB storage, and 100 MB max file size. Could you clarify if these restrictions are enforced by proprietary code in the AFFiNE Cloud components or if they’re purely configuration-based and could be modified in a fully self-hosted setup? For example, can the community adjust these limits (e.g., member count or storage) by tweaking the open-source code, or are there closed parts involved?

Just trying to better understand the boundaries of the open-source version for self-hosting. Thanks for any insights!

**Reply by CREAsTIVE** [reply]
Same question, but it looks like its illegal to unlock them through source code tweeking

**Reply by bernie43** [reply]
I'm not a lawyer and am not affiliated with Affine, but the license for the server code sounds pretty clear to me: https://github.com/toeverything/AFFiNE/blob/canary/packages/backend/server/LICENSE 

```
This software and associated documentation files (the "Software") may only be
used in production, if you (and any entity that you represent) have agreed to,
and are in compliance with, the AFFiNE Subscription Terms of Service [...]
and otherwise have a valid AFFiNE Enterprise Edition subscription for the
correct number of user seats.
```

So to run the server in production you need to comply with the terms, and they say it's free for up to 10 users and 100 GB of storage, otherwise you need to pay per seat.
Since the server code is available, no one could be stopped with any technical measures from removing the limits, but it would be against the license terms.
Apart from removing the limits, from what I understand you can modify and extend the server code as much as you like. `you are free to modify this Software and publish patches to the Software`

(Everything else wouldn't make sense anyway, since Affine is a for-profit company, and their business model seems to be to make money by selling to enterprises)

**Comment by jachane99** [comment]
For anyone who comes across this thread and considers modifications for their organizations: none of the code is actually under the MIT license, which is misleading. Even the frontend code, which claims to be MIT, actually uses Octobase https://github.com/toeverything/OctoBase and is licensed under [AGPL-3.0 license](https://github.com/toeverything/OctoBase#AGPL-3.0-1-ov-file). Since Affine directly uses this package, it becomes "infected" with the restricted license. Therefore, any changes you make must be made public. Not sure why this is hidden and they don't want to make it obvious and clarify their license terms.

**Reply by bernie43** [reply]
I'm pretty sure they don't use Octobase anymore. The local version uses Sqlite as database, and the server uses Postgres.
Can you post any code references?

**Reply by jachane99** [reply]
omg I have never been happier to be wrong. I was not up to date with the package and was looking at some old references. Thank you kind stranger for correcting me.
